Automattic/woocommerce-payments
# Connect the TOS modal

#### Changes proposed in this Pull Request

Builds on top of https://github.com/Automattic/woocommerce-payments/pull/983, connects the ToS modal to the server (PR 404-gh-woocommerce-payments-server), and adds the rest of the necessary functionality for the ToS modal:

1. Move the ToS modal from `client/tos.js` to `client/tos/index.js`.
2. Add erroneous messages/states to the modal.
3. Add the `client/tos/snackbar/index.js` snackbar, which appears as a notice once the gateway has been disabled and allows the action to be undone.
4. Record the acceptance in `WC_REST_Payments_Tos_Controller::handle_tos_accepted` and add the `reactivate` method to the class for the snackbar.
5. Modify `WC_Payments_Admin` to check whether a ToS acceptance is required when enqueuing the ToS script.


#### Testing instructions

1. Check out the server PR as well as this one, run `npm run build:client`.
2. Navigate to the gateway settings page. You should see the modal.
3. Accept the ToS. The modal should disappear and not be visible anymore. To confirm the flag, navigate to dev tools and makre sure that `'is_current_version' => true,` is present in the account cache contents.
4. Delete the record from the server table and clear the local account cache.
5. Navigate to the gateway settings again and you should see the modal.
6. This time, decline the ToS and disable the gateway in the second step.
7. For the first 15 seconds you should see a snackbar, which says the gateway got disabled and allow you to undo the action. Click the "Undo" button.
8. The gateway should be active again and you should see the gateway settings page with the modal on top of it.

-------------------

- [ ] Figure out exactly when to show the modal before merging.
